stbt_control_relay.py

```diff
@@ -68,47 +69,45 @@ def main(argv):
 
     while True:
         conn, _ = s.accept()
-        f = conn.makefile('r', 0)
+        f = conn.makefile('rb', 0)
         while True:
             cmd = f.readline()
             if not cmd:
                 break
-            cmd = cmd.rstrip("\n")
-            m = re.match(r"(?P<action>SEND_ONCE|SEND_START|SEND_STOP) "
-                         r"(?P<ctrl>\S+) (?P<key>\S+)", cmd)
+            cmd = cmd.rstrip(b"\n")
+            m = re.match(br"(?P<action>SEND_ONCE|SEND_START|SEND_STOP) "
+                         br"(?P<ctrl>\S+) (?P<key>\S+)", cmd)
             if not m:
                 debug("Invalid command: %s" % cmd)
                 send_response(conn, cmd, success=False,
-                              data="Invalid command: %s" % cmd)
+                              data=b"Invalid command: %s" % cmd)
                 continue
-            action = m.groupdict()["action"]
-            key = m.groupdict()["key"]
+            action = m.group("action")
+            key = m.group("key")
             debug("Received %s %s" % (action, key))
             try:
-                if action == "SEND_ONCE":
+                if action == b"SEND_ONCE":
                     control.press(key)
-                elif action == "SEND_START":
+                elif action == b"SEND_START":
                     control.keydown(key)
-                elif action == "SEND_STOP":
+                elif action == b"SEND_STOP":
                     control.keyup(key)
             except Exception as e:  # pylint: disable=broad-except
                 debug("Error pressing key %r: %r" % (key, e))
-                send_response(conn, cmd, success=False, data=str(e))
+                send_response(conn, cmd, success=False, data=to_bytes(str(e)))
```

Comment:
On Python 2 this will call

    e.__unicode__().encode('utf-8')

On Python 3 will be

    e.__str__().encode('utf-8')

Looks fine.